### PR TITLE
US004 view band level

### PR DESCRIPTION
**Pull request by Criofann**
As a Kainos employee
I want to see what band level each role falls within
So that I can see what level the role is within Kainos corporate structure

       - New endpoint for band level lookup successfully added and double checked on swagger.
       - Full DAO and Service added for the Band table - including all fields for future use

Screenshot 2023-12-06 at 14 11 37

Screenshot 2023-12-06 at 14 12 32

### CODE REVIEW

**Review by Joseph-Magee — DISMISSED** [review]
looks good but merge conflicts need fuxed

**Comment by Criofann** [comment]
@jkelso662 @lullabelle Merge conflicts resolved and peer approved need one of you guys to check over please

**Review by Joseph-Magee — APPROVED** [review]
looks good

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -11,8 +11,7 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout code
```
Comment (jkelso662) [inline .github/workflows/maven.yml:14]: Is there a reason yous got rid of the name of this first step?

File: src/main/java/org/kainos/ea/cli/Band.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.kainos.ea.cli;
+
+public class Band {
+    private String BandName;
+    private String BandLevel;
+    private String Competencies;
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/cli/Band.java:6]: Thought we agreed these should start with lowercase? Very unusual! Makes these look like class references!

File: src/main/java/org/kainos/ea/cli/BandRole.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package org.kainos.ea.cli;
+
+public class BandRole {
+    private String RoleName;
+    private String Specification;
+    private String Responsibilities;
+    private String SharepointLink;
+    private String BandLevel;
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/cli/BandRole.java:8]: Same here

File: src/main/java/org/kainos/ea/cli/RoleRequest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class RoleRequest {
+    private String roleName;
+    private String specification;
+    private String capabilityName;
+    private String bandName;
+    private String responsibilities;
+    private String sharepointLink;
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/cli/RoleRequest.java:12]: I.e. these are spot on 💯

File: src/main/java/org/kainos/ea/core/RoleValidator.java
Comment (jkelso662) [inline src/main/java/org/kainos/ea/core/RoleValidator.java:1]: ❤️

**Review by jkelso662 — CHANGES_REQUESTED** [review]
Heyy guys, not too bad!  Few things to sort tho.
1. Main this is why have you removed your linter?
2. I don't see any automated tests?
3. There is a funny capitalisation thing going on in them Java classes we discussed back in Java week

File: .github/workflows/maven.yml [hunk 5]
```diff
@@ -22,7 +21,4 @@ jobs:
           cache: maven
 
       - name: Build with Maven
-        run: mvn clean install
-
-      - name: Run linter
-        run: mvn checkstyle:check
```
[the remark this answers is not in the document]
Comment (Criofann) [reply .github/workflows/maven.yml:28]: linter was changed so deleting the old one

**Comment by Criofann** [comment]
@jkelso662 @lullabelle all points brought up have been fixed and passing all linter checks too

File: src/main/java/org/kainos/ea/client/BandDoesNotExistException.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package org.kainos.ea.client;
+
+public class BandDoesNotExistException extends Throwable {
+    @Override
+    public String getMessage() {
+        return
+                "Band selected does not exist";
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/client/BandDoesNotExistException.java:7]: This is very weird spacing! haha Why use two lines when you could use one?

File: src/main/java/org/kainos/ea/cli/Band.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    public void setCompetencies(String competencies) {
+        this.competencies = competencies;
+    }
+
+    public Band(String bandName,
+                String bandLevel, String competencies) {
+        this.bandName = bandName;
+        this.bandLevel = bandLevel;
+        this.competencies = competencies;
+    }
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/cli/Band.java:37]: Constructor seems weird down here, usually above getters and setters.

**Review by jkelso662 — APPROVED** [review]
Have a look at the comments I left but otherwise looks good :)

File: src/main/java/org/kainos/ea/client/BandDoesNotExistException.java (hunk 6 above)
Comment (Criofann) [reply src/main/java/org/kainos/ea/client/BandDoesNotExistException.java]: Believe it was a linter error I was trying to fix and it messed with the formatting

**Comment by Criofann** [comment]
> Have a look at the comments I left but otherwise looks good :)

All fixed! Just some formatting issues I must have missed whenever fixing linter checks

**Pull request closed without merging**